scrapy/pipelines/files.py

```diff
@@ -232,6 +233,7 @@ def from_settings(cls, settings):
         s3store = cls.STORE_SCHEMES['s3']
         s3store.AWS_ACCESS_KEY_ID = settings['AWS_ACCESS_KEY_ID']
         s3store.AWS_SECRET_ACCESS_KEY = settings['AWS_SECRET_ACCESS_KEY']
+        s3store.POLICY = settings['S3_STORE_ACL']
```

Comment:
The PR proposes to change policy of objects uploaded by feed exporters, this pipeline is not used for that.

You should look at scrapy/extensions/feedexporters.py for S3FeedStorage instead.